### ISSUE TITLE
transitioning from 5D & 4D layers to only 4D layer throws error

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#671]
## 🐛 Bug

Transitioning from having two layers, one 5D and one 4D to having just one 4D layer by deleting the 5D one causes an error, likely introduced during the grid mode work. we need to fix and add a test.

```
WARNING: Traceback (most recent call last):
  File "/Users/nicholassofroniew/Github/napari/napari/_qt/qt_viewer_buttons.py", line 57, in <lambda>
    self.clicked.connect(lambda: self.viewer.layers.remove_selected())
  File "/Users/nicholassofroniew/Github/napari/napari/components/layerlist.py", line 126, in remove_selected
    self.pop(i)
  File "/Users/nicholassofroniew/Github/napari/napari/util/list/_model.py", line 60, in pop
    self.events.removed(item=obj, index=key)
  File "/Users/nicholassofroniew/Github/napari/napari/util/event.py", line 508, in __call__
    self._invoke_callback(cb, event)
  File "/Users/nicholassofroniew/Github/napari/napari/util/event.py", line 529, in _invoke_callback
    cb_event=(cb, event),
  File "/Users/nicholassofroniew/Github/napari/napari/util/event.py", line 523, in _invoke_callback
    cb(event)
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 99, in <lambda>
    self.layers.events.removed.connect(lambda e: self._update_grid())
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 1191, in _update_grid
    stride=self.grid_stride,
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 1178, in grid_view
    self._subplot(layer, (i_row, i_column))
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 1206, in _subplot
    scene_size, corner = self._scene_shape()
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 267, in _scene_shape
    size = [size[i] for i in self.dims.displayed]
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 267, in <listcomp>
    size = [size[i] for i in self.dims.displayed]
IndexError: index 4 is out of bounds for axis 0 with size 4
```

**Comment by sofroniewn** [comment]
This is a duplicated of #1323 and will be closed by #1359

**Comment by jni** [comment]
Closed by #1359
